### PR TITLE
Y2logs rewrite

### PR DESCRIPTION
**Pull request by wfeldt**

### CODE REVIEW

File: scripts/save_y2logs [hunk 1]
```diff
@@ -7,110 +7,140 @@
...
+	;;
+
     *.tgz|*.tar.gz)
 	TARGET="$1"
-	COMPRESSION=--gzip
+	COMPRESSION=gzip
 	;;
 
-    *.tar.bz2)
+    *.tar.xz)
```
Comment (jreidinger) [inline scripts/save_y2logs]: I also see usage of `txz` so maybe also support it?

File: scripts/save_y2logs [hunk 2]
```diff
@@ -7,110 +7,140 @@
...
-	COMPRESSION=--bzip2
+	COMPRESSION=xz
+	;;
 
-	if [ ! -x /usr/bin/bzip2 ]; then
-	    echo "FATAL: /usr/bin/bzip2 not available" >&2
-	    # This might easily happen in the inst-sys
-            exit 3
-	fi
+    *.tar.bz2)
```
Comment (jreidinger) [inline scripts/save_y2logs]: same here with `tbz2` as shortcut.

File: scripts/save_y2logs [hunk 3]
```diff
@@ -7,110 +7,140 @@
...
+	TARGET="$1"
+	COMPRESSION=bzip2
 	;;
 
     *)
-	echo "FATAL: Uncompressed archives not supported" >&2
-	echo "Use one of: .tgz .tar.gz .tar.bz2"          >&2
+	echo "FATAL: argument is not a (compressed) tar file." >&2
+	echo "Use one of these suffixes: .tar .tgz .tar.gz .tar.xz .tar.bz2" >&2
 	exit 4
```
Comment (jreidinger) [inline scripts/save_y2logs]: target in all reasonable branches is always same TARGET=$1, maybe it make sense to assign it outside of case statement.

File: scripts/save_y2logs [hunk 4]
```diff
@@ -7,110 +7,140 @@
...
   esac
 fi
 
+# check for compression program
+if [ "$COMPRESSION" -a ! -x "/usr/bin/$COMPRESSION" ]; then
```
Comment (jreidinger) [inline scripts/save_y2logs]: can we be a bit robust with `which $COMPRESSION` for location of compression algorithm? Who know when someone decide to drop /usr or move it to some funny location.

File: scripts/save_y2logs [hunk 5]
```diff
@@ -7,110 +7,140 @@
...
-if [ -x /bin/dmesg ]; then
-    dmesg > $TMPDIR/dmesg
-    LIST="$LIST $TMPDIR/dmesg"
-fi
+# 1. gather some information in $TMPDIR
 
-if [ -f /var/log/zypp/history ]; then
-    LIST="$LIST $( cd /var/log/; ls zypp/history*)"
+# last kernel messages, if any
+if [ -x /bin/dmesg ]; then
```
Comment (jreidinger) [inline scripts/save_y2logs]: what about `journalctl --dmesg` for journalctl? it can be useful on running system.

File: scripts/save_y2logs [hunk 6]
```diff
@@ -7,110 +7,140 @@
...
+VAR_LOG_FILES='\
+  YaST2 \
+  Xorg.0.log \
+  zypper.log zypp/history* pk_backend_zypp \
+  pbl.log linuxrc.log wickedd.log \
+  evms-engine.* \
+'
+cd /var/log
+for i in $VAR_LOG_FILES ; do
+  [ -e "$i" ] && LIST_VAR="$LIST_VAR $i"
```
Comment (jreidinger) [inline scripts/save_y2logs]: this does not work with asterisk. Try e.g.

```
jreidinger@pepa:/tmp> ls /tmp/y2log*
/tmp/y2log-1.gz.swp  /tmp/y2logs.tgz
jreidinger@pepa:/tmp> [ -e "/tmp/y2log*" ]
jreidinger@pepa:/tmp> echo $?
1
jreidinger@pepa:/tmp> [ -e "/tmp/y2logs.tgz" ]
jreidinger@pepa:/tmp> echo $?
0
```

File: scripts/save_y2logs (hunk 2 above)
Comment (wfeldt) [inline scripts/save_y2logs]: Uhg, I can go with txz, but tbz2 is a bit over-the-top. bzip2 is quite dead.

File: scripts/save_y2logs (hunk 6 above)
Comment (wfeldt) [inline scripts/save_y2logs]: it does work, $VAR_LOG_FILES gets expanded

File: scripts/save_y2logs (hunk 5 above)
Comment (wfeldt) [inline scripts/save_y2logs]: just tried and it's quite big (120MB on my machine); xz deals wiith it, though...

File: scripts/save_y2logs (hunk 6 above)
Comment (jreidinger) [inline scripts/save_y2logs]: ah, dark bash magic, with having it in `''` and then expanded for cycle. OK, I trust to verify it.

File: scripts/save_y2logs (hunk 5 above)
Comment (jreidinger) [inline scripts/save_y2logs]: point is that it is replacement for dmesg utility itself. At least on my pc, both return same output. Now it do not make much sense to have both, as both should return same lines.

**Comment by jreidinger** [comment]
it looks good now, just we should store dmesg or journalctl --dmesg, not both as it have same content see https://github.com/yast/yast-yast2/pull/415#discussion_r45743809 just we should try both if any of it is available, then use it.

**Comment by wfeldt** [comment]
dmesg and journalctrl --dmesg are not the same on my machine

**Comment by jreidinger** [comment]
ok, then it make sense having both. LGTM

**Pull request merged**